core/unit_test/TestLocalDeepCopy.hpp

```diff
@@ -14,894 +14,272 @@
-      });
+template <typename ViewType, typename Bounds>
+KOKKOS_INLINE_FUNCTION auto extract_subview(
+    ViewType& src, int lid, Bounds bounds,
+    std::enable_if_t<unsigned(ViewType::rank) == 4>* = nullptr) {
+  return Kokkos::subview(src, lid, bounds, Kokkos::ALL, Kokkos::ALL);
+}
 
-  Kokkos::deep_copy(h_A, A);
-  Kokkos::deep_copy(h_B, B);
+template <typename ViewType, typename Bounds>
+KOKKOS_INLINE_FUNCTION auto extract_subview(
+    ViewType& src, int lid, Bounds bounds,
+    std::enable_if_t<unsigned(ViewType::rank) == 5>* = nullptr) {
+  return Kokkos::subview(src, lid, bounds, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL);
+}
 
-  bool test = true;
-  for (size_t i = 0; i < A.span(); i++) {
-    if (h_A.data()[i] != h_B.data()[i]) {
-      test = false;
-      break;
-    }
-  }
+template <typename ViewType, typename Bounds>
+KOKKOS_INLINE_FUNCTION auto extract_subview(
+    ViewType& src, int lid, Bounds bounds,
+    std::enable_if_t<unsigned(ViewType::rank) == 6>* = nullptr) {
+  return Kokkos::subview(src, lid, bounds, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL, Kokkos::ALL);
+}
 
-  ASSERT_EQ(test, true);
+template <typename ViewType, typename Bounds>
+KOKKOS_INLINE_FUNCTION auto extract_subview(
+    ViewType& src, int lid, Bounds bounds,
+    std::enable_if_t<unsigned(ViewType::rank) == 7>* = nullptr) {
+  return Kokkos::subview(src, lid, bounds, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL, Kokkos::ALL, Kokkos::ALL);
+}
 
-  // Fill
-  Kokkos::deep_copy(B, 0.0);
+template <typename ViewType, typename Bounds>
+KOKKOS_INLINE_FUNCTION auto extract_subview(
+    ViewType& src, int lid, Bounds bounds,
+    std::enable_if_t<unsigned(ViewType::rank) == 8>* = nullptr) {
+  return Kokkos::subview(src, lid, bounds, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL, Kokkos::ALL, Kokkos::ALL, Kokkos::ALL);
+}
 
-  Kokkos::parallel_for(
-      team_policy(N, Kokkos::AUTO),
-      KOKKOS_LAMBDA(const member_type& teamMember) {
-        int lid = teamMember.league_rank();  // returns a number between 0 and N
-        auto subDst = Kokkos::subview(B, 1, 1, 1, 1, lid, Kokkos::ALL(),
-                                      Kokkos::ALL(), Kokkos::ALL());
-        Kokkos::Experimental::local_deep_copy(teamMember, subDst, 20.0);
-      });
+template <typename ViewType>
+void reset(ViewType B) {
+  Kokkos::deep_copy(B, 0);
+}
 
-  Kokkos::deep_copy(h_B, B);
+template <typename ViewType>
+bool check_sum(ViewType B, const int N,
+               typename ViewType::value_type fill_value) {
+  double sum_all = 0;
+  Kokkos::parallel_reduce(
+      "Check B", B.span(),
+      KOKKOS_LAMBDA(int i, double& lsum) { lsum += B.data()[i]; },
+      Kokkos::Sum<double>(sum_all));
 
-  double sum_all = 0.0;
-  for (size_t i = 0; i < B.span(); i++) {
-    sum_all += h_B.data()[i];
+  auto correct_sum = fill_value;
+  for (auto i = 0; i < ViewType::rank; i++) {
```

Comment:
You are getting warnings about comparing signed and unsigned types.
```suggestion
  for (decltype(ViewType::rank) i = 0; i < ViewType::rank; ++i) {
```